pkg/clarity/types.go

```diff
@@ -0,0 +1,658 @@
+	return bool(b)
+}
+
+type StandardPrincipal struct {
+	Version byte
+	Hash160 [20]byte
+}
+
+func (p *StandardPrincipal) Serialize() ([]byte, error) {
+	buf := make([]byte, 22)
+	buf[0] = byte(ClarityTypeStandardPrincipal)
+	buf[1] = p.Version
+	copy(buf[2:], p.Hash160[:])
+	return buf, nil
+}
+
+func NewStandardPrincipal(version byte, hash160 [20]byte) *StandardPrincipal {
+	return &StandardPrincipal{Version: version, Hash160: hash160}
+}
+
+func (p *StandardPrincipal) Type() ClarityType {
+	return ClarityTypeStandardPrincipal
+}
+
+type ContractPrincipal struct {
+	StandardPrincipal
+	ContractName string
+}
+
+func NewContractPrincipal(version byte, hash160 [20]byte, contractName string) (*ContractPrincipal, error) {
+	if len(contractName) > 128 {
+		return nil, fmt.Errorf("contract name too long (max 128 characters)")
+	}
+	return &ContractPrincipal{
+		StandardPrincipal: StandardPrincipal{Version: version, Hash160: hash160},
+		ContractName:      contractName,
+	}, nil
+}
+
+func (p *ContractPrincipal) Type() ClarityType {
+	return ClarityTypeContractPrincipal
+}
+
+func (p *ContractPrincipal) Serialize() ([]byte, error) {
+	result := []byte{byte(ClarityTypeContractPrincipal), p.Version}
+	result = append(result, p.Hash160[:]...)
+	result = append(result, byte(len(p.ContractName)))
+	result = append(result, []byte(p.ContractName)...)
+	return result, nil
+}
+
+type ResponseOk struct {
+	Value ClarityValue
+}
+
+func NewResponseOk(value ClarityValue) *ResponseOk {
+	return &ResponseOk{Value: value}
+}
+
+func (r *ResponseOk) Type() ClarityType {
+	return ClarityTypeResponseOk
+}
+
+func (r *ResponseOk) Serialize() ([]byte, error) {
+	valueBytes, err := r.Value.Serialize()
+	if err != nil {
+		return nil, err
+	}
+	return append([]byte{byte(ClarityTypeResponseOk)}, valueBytes...), nil
+}
+
+type ResponseErr struct {
+	Value ClarityValue
+}
+
+func NewResponseErr(value ClarityValue) *ResponseErr {
+	return &ResponseErr{Value: value}
+}
+
+func (r *ResponseErr) Type() ClarityType {
+	return ClarityTypeResponseErr
+}
+
+func (r *ResponseErr) Serialize() ([]byte, error) {
+	valueBytes, err := r.Value.Serialize()
+	if err != nil {
+		return nil, err
+	}
+	return append([]byte{byte(ClarityTypeResponseErr)}, valueBytes...), nil
+}
+
+type OptionNone struct{}
+
+func NewOptionNone() *OptionNone {
+	return &OptionNone{}
+}
+
+func (o *OptionNone) Type() ClarityType {
+	return ClarityTypeOptionNone
+}
+
+func (o *OptionNone) Serialize() ([]byte, error) {
+	return []byte{byte(ClarityTypeOptionNone)}, nil
+}
+
+type OptionSome struct {
+	Value ClarityValue
+}
+
+func NewOptionSome(value ClarityValue) *OptionSome {
+	return &OptionSome{Value: value}
+}
+
+func (o *OptionSome) Type() ClarityType {
+	return ClarityTypeOptionSome
+}
+
+func (o *OptionSome) Serialize() ([]byte, error) {
+	valueBytes, err := o.Value.Serialize()
+	if err != nil {
+		return nil, err
+	}
+	return append([]byte{byte(ClarityTypeOptionSome)}, valueBytes...), nil
```

Comment:
In other files you followed space after bracket formatting. Might be worth to be consistent.

Reply:
I'm not sure I follow.

<img width="494" alt="image" src="https://github.com/user-attachments/assets/a1e4f88e-d8ad-43cb-b3fc-023c927affc7">

Reply:
Aah just formatting that after a bracket there is a new line
like 

}

"next here"